### PR TITLE
Implement min, max, sum and mean of Rolling in Series and DataFrame

### PR DESCRIPTION
**Pull request by itholic**
This PR implement min, max, sum and mean of Rolling in Series and DataFrame

For Series:

```python
>>> kser = ks.Series([2, 3, float("nan"), 10])
>>> kser
0     2.0
1     3.0
2     NaN
3    10.0
Name: 0, dtype: float64

>>> ksr.rolling(2).sum()
0    NaN
1    5.0
2    3.0
3    NaN
Name: 0, dtype: float64

>>> ks.rolling(2).min()
0    NaN
1    2.0
2    3.0
3    NaN
Name: 0, dtype: float64

>>> ks.rolling(2).max()
0    NaN
1    3.0
2    3.0
3    NaN
Name: 0, dtype: float64

>>> ks.rolling(2).mean()
0    NaN
1    2.5
2    3.0
3    NaN
Name: 0, dtype: float64
```

For DataFrame

```python
>>> kdf = ks.DataFrame({'a': [1, float('nan'), 3], 'b': [1.0, 2.0, 3.0]})
>>> kdf
     a    b
0  1.0  1.0
1  NaN  2.0
2  3.0  3.0

>>> kdf.rolling(2).sum()
     a    b
0  NaN  NaN
1  1.0  3.0
2  NaN  5.0

>>> kdf.rolling(2).min()
     a    b
0  NaN  NaN
1  1.0  1.0
2  NaN  2.0

>>> kdf.rolling(2).max()
     a    b
0  NaN  NaN
1  1.0  2.0
2  NaN  3.0

>>> kdf.rolling(2).mean()
     a    b
0  NaN  NaN
1  1.0  1.5
2  NaN  2.5
```

relates to #977

### CODE REVIEW

File: databricks/koalas/window.py [hunk 1]
```diff
@@ -42,6 +60,369 @@ def __getattr__(self, item: str) -> Any:
...
+
+        >>> df.rolling(3).mean()
+                  A          B
+        0       NaN        NaN
+        1       NaN        NaN
+        2  4.000000  16.666667
+        3  3.333333  12.666667
+        4  4.333333  21.666667
+        """
+        def mean(scol, index_scols):
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: Why do we need to send `index_scols`? Seems it can be set within `_apply_as_series_or_frame`.

Comment (itholic) [reply databricks/koalas/window.py]: ah, you're right, i'll fix them and also https://github.com/databricks/koalas/pull/990#discussion_r342359478

Comment (itholic) [reply databricks/koalas/window.py]: Thanks for the comment! i fixed them

File: databricks/koalas/window.py [hunk 2]
```diff
@@ -42,6 +60,369 @@ def __getattr__(self, item: str) -> Any:
...
+        >>> df.rolling(3).sum()
+              A     B
+        0   NaN   NaN
+        1   NaN   NaN
+        2  12.0  50.0
+        3  10.0  38.0
+        4  13.0  65.0
+        """
+        def sum(scol):
+            window = Window.orderBy(self._index_scols)
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: Can we move this into `_apply_as_series_or_frame` and use `self._window`? We can match the implementation into Expanding one I think.

File: databricks/koalas/window.py [hunk 3]
```diff
@@ -30,8 +30,26 @@ class _RollingAndExpanding(object):
 
 
 class Rolling(_RollingAndExpanding):
-    def __init__(self, obj):
-        self.obj = obj
+    def __init__(self, kdf_or_kser, window, min_periods=None):
+        from databricks.koalas import DataFrame, Series
+        from databricks.koalas.groupby import SeriesGroupBy, DataFrameGroupBy
+        window = window - 1
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: Why do we do `-1` and `+1` when it's actually used via `_window_val`?

File: databricks/koalas/window.py [hunk 4]
```diff
@@ -42,6 +60,369 @@ def __getattr__(self, item: str) -> Any:
                 return partial(property_or_func, self)
         raise AttributeError(item)
 
+    def _apply_as_series_or_frame(self, func):
+        """
+        Decorator that can wraps a function that handles Spark column in order
```
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: `Decorator that can wraps` -> `Wraps`

File: databricks/koalas/window.py (hunk 3 above)
Comment (HyukjinKwon) [inline databricks/koalas/window.py]: I wouldn't do this but do `-1` or `+1` when it's actually used later.

File: databricks/koalas/generic.py [hunk 5]
```diff
@@ -1393,8 +1393,9 @@ def median(self, accuracy=10000):
         # This is expected to be small so it's fine to transpose.
         return DataFrame(sdf)._to_internal_pandas().transpose().iloc[:, 0]
 
-    def rolling(self, *args, **kwargs):
-        return Rolling(self)
+    # TODO: 'center', 'win_type', 'on', 'axis' parameter should be implemented.
+    def rolling(self, window, min_periods=None):
```
Comment (HyukjinKwon) [inline databricks/koalas/generic.py]: @itholic can we add a doc here or in https://github.com/databricks/koalas/pull/990?

File: databricks/koalas/tests/test_rolling.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            ks.range(10).rolling(window=-1)
+        with self.assertRaisesRegex(ValueError, "min_periods must be >= 0"):
+            ks.range(10).rolling(window=1, min_periods=-1)
+
+        with self.assertRaisesRegex(
+                TypeError,
+                "kdf_or_kser must be a series or dataframe; however, got:.*int"):
+            Rolling(1, 2)
+
+    def _test_rolling_func(self, f):
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_rolling.py]: I added some tests for multiindex at https://github.com/databricks/koalas/commit/94f5ab48905b5c99dfa2d153b96579ffed89796d. you can add it here as well.

File: databricks/koalas/generic.py (hunk 5 above)
Comment (itholic) [reply databricks/koalas/generic.py]: okay i'll add :)

File: databricks/koalas/tests/test_rolling.py (hunk 6 above)
Comment (itholic) [reply databricks/koalas/tests/test_rolling.py]: yes sir 😄 thanks for the comment!

File: databricks/koalas/window.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/window.py]: is it possible although this `window` and `self._window` are used for different purpose like below?

스크린샷 2019-11-06 오후 3 18 34

**Comment by HyukjinKwon** [comment]
@itholic can you rebase please?

**Comment by itholic** [comment]
@HyukjinKwon sure, i just did.

**Comment by HyukjinKwon** [comment]
Let me clean up by myself. merged

**Comment by itholic** [comment]
@HyukjinKwon Thanks :)

**Pull request merged**